src/segment_type.rs

```diff
@@ -69,9 +69,27 @@ impl TryFrom<u8> for SegmentType {
     }
 }
 
+impl FromStr for SegmentType {
+    type Err = ReadStructureError;
+
+    /// Returns the segment type given the string representation.
+    ///
+    /// # Errors
+    ///
+    /// - If `SegmentType` not valid
+    fn from_str(value: &str) -> Result<Self, Self::Err> {
+        if value.len() == 1 {
+            Self::try_from(value.chars().nth(0).unwrap())
+        } else {
+            Err(ReadStructureError::ReadSegmentMultipleSegments(value.to_owned()))
```

Comment:
I think we use `ReadSegmentTypeInvalid` or we need a new error type for when the segment type is mis-specified (empty string or multiple characters).

Reply:
`ReadSegmentTypeInvalid` takes a `char`, so added a new (but very similar) error type `ReadSegmentTypeStringInvalid` for when the segment type if empty or multiple characters along with the relevant tests.